app/models/user.rb

```diff
@@ -550,6 +550,26 @@ def close_account!
     AccountDeletion.create(:person => self.person)
   end
 
+  def lock_access!(opts={})
+    if opts[:unlock_in]
+      self.lock_expires = true
+      self.locked_at = Time.now.utc + opts[:unlock_in]
+      save
+    else
+      self.lock_expires = false
+      super
+    end
+  end
+
+  def lock_expired?
```

Comment:
Have you checked that this actually gets called? The [devise docs](http://www.rubydoc.info/github/plataformatec/devise/Devise/Models/Lockable) say this only gets called if the lock strategy is `time`, which is not the case for us.